test/de/willuhn/jameica/hbci/server/TestVerwendungszweckUtil.java

```diff
@@ -2,315 +2,679 @@
+      };
+
+      Map<Tag, String> map = VerwendungszweckUtil.parse(test);
+      check(map, "SVWZ", "Das folgende Tag gibts nichtFooo ABCD+Gehoert zum Verwendungszweck");
+      check(map, "EREF", "Aber hier kommt noch was");
+    }
+
+    @Test
+    void leerzeichenHinterTagWerdenIgnoriert() throws Exception
+    {
+      String[] test = {
+              "SVWZ+Das folgende Tag gibts nicht",
+              "Fooo ",
+              "ABCD+ Leerzeichen hinter dem Tag stoeren nicht",
+              "KREF+ und hier stoeren sie auch nicht, sind aber nicht teil des Value "
+      };
+
+      Map<Tag, String> map = VerwendungszweckUtil.parse(test);
+      check(map, "SVWZ", "Das folgende Tag gibts nichtFooo ABCD+ Leerzeichen hinter dem Tag stoeren nicht");
+      check(map, "KREF", "und hier stoeren sie auch nicht, sind aber nicht teil des Value");
+    }
+
+    @Test
+    void gemischtePlusUndDoppelpunktWerdenErkannt() throws Exception
+    // TODO hier Code von #parse() anpassen, um noch besser zu trennen? Was ist vom Anwender gewollt?
+    {
+      String[] test = {
+              "SVWZ+ Das geht sogar",
+              " gemischt ",
+              "IBAN: DE1234567890 ",
+              "BIC: AACSDE33"
+      };
+
+      Map<Tag, String> map = VerwendungszweckUtil.parse(test);
+      // ACHTUNG: durch die Mischung sind die Anteile mit Doppelpunkt im SVWZ enthalten!
+      check(map, "SVWZ", "Das geht sogar gemischt IBAN: DE1234567890 BIC: AACSDE33");
+      check(map, "IBAN", "DE1234567890");
+      check(map, "BIC", "AACSDE33");
+    }
+
+    //region Sonderrolle IBAN und BIC
+
+    /**
+     * ehemals "testParse012"
+     *
+     * @see #gemischtePlusUndDoppelpunktWerdenErkannt()
+     */
+    @Test
+    void beiMischungVonPlusUndDoppelpunktSindAllePunktTexteImTagMitPlusEnthalten() throws Exception
+    {
+      // TODO: hier Code von #parse() anpassen, um noch besser zu trennen? Was ist vom Anwender gewollt?
+      // Aus Code:
+      // Sonderrolle IBAN. Wir entfernen alles bis zum ersten Leerzeichen. Siehe "testParse012". Da hinter der
+      // IBAN kein vernuenftiges Tag mehr kommt, wuerde sonst der ganze Rest da mit reinfallen. Aber nur, wenn
+      // es erst nach 22 Zeichen kommt. Sonst steht es mitten in der IBAN drin. In dem Fall entfernen wir die
+      // Leerzeichen aus der IBAN (siehe "testParse013")
+      // TODO: Ist IBAN per Definition stets das letzte Tag? ("da hinter der IBAN kein vern�nftiges Tag mehr kommt")
+      String[] test = {
+              "SVWZ+BIC:GENODED1SAM ",
+              "IBAN:DE12345678901234567890 ",
+              "Datum: 14.01.16 Zeit: 08:00 ",
+              "KD 00012345 TAN 12345 ",
+              "Beleg 12345  Kunde 12345"
+      };
+      // ACHTUNG: durch die Mischung sind die Anteile mit Doppelpunkt im SVWZ enthalten!
+      String svwz = "BIC:GENODED1SAM IBAN:DE12345678901234567890 Datum: 14.01.16 Zeit: 08:00 KD 00012345 TAN 12345 Beleg 12345  Kunde 12345";
```

Comment:
Wäre es nicht besser, `+` und `:` äquivalent zu behandeln? Dann wäre der Text von dem Einen nicht mehr Bestandteil vom Anderen... (siehe auch `gemischtePlusUndDoppelpunktWerdenErkannt ()`)

Hier wäre dann folglich zu überlegen, was dann Bestandteil von SVWZ wird. BIC und IBAN werden erkannt. Sollte der Rest dann in SVWZ bleiben (also alles vor dem Tag BIC und alles nach IBAN)? 
Hier wäre mein Vorschlag:
```
BIC = "GENODED1SAM"
IBAN = "DE12345678901234567890"
SVWZ = "Datum: 14.01.16 Zeit: 08:00 KD 00012345 TAN 12345 Beleg 12345  Kunde 12345"
```

Oder ist das fachlich nicht sinnvoll?